modules/solid_mechanics/src/kernels/StressDivergenceRZ.C

```diff
@@ -41,6 +43,21 @@ StressDivergenceRZ::StressDivergenceRZ(const InputParameters & parameters)
 Real
 StressDivergenceRZ::computeQpResidual()
 {
+  // calculate volume averaged value of shape function derivative
+  Real Avg_derivative_0 = 0.0; // Average shape function derivatives in direction 0
+  Real Avg_derivative_1 = 0.0; // Average shape function derivatives in direction 1
```

Comment:
This should be formulated using a RealVector value. Then you do not need to do component wise math below.